### ISSUE TITLE
FEATURE request: Add support for TLS/SSL

### ISSUE THREAD

**Issue by paladox** [moosefs/moosefs#398]
## Have you read through available documentation and open Github issues?

Yes

## Is this a BUG report, FEATURE request, or a QUESTION? Who is the indended audience?

FEATURE request

## Information

We currently use GlusterFS but would like to move to MooseFS but our only issue is it doesn't support TLS/SSL. We've previously used LizardFS (which was performant, which is why I presume moosefs will be too) but moved to GlusterFS which has TLS/SSL. Could this support be added please (in the chunkservers/master/shadow & client)?

**Comment by chogata** [comment]
We do have encryption of connections on our roadmap. However a) it will have impact on performance, there is no doubt, tests will show how much of an issue it will be; it will never be a default, just an option; b) we plan encryption with pre-defined keys, not SSL. We wonder, if SSL is really necessary. It would be interested to hear from our community - why EXACTLY do you need encrypted connections?

**Comment by jkiebzak** [comment]
why not use WireGuard?

>

**Comment by protosam** [comment]
Sometimes compliance requires the individual services in-network to be encrypted.

I recommend just going with openssl @chogata , it's already solved, and people get to pick their own ciphers for their compliance needs.

This issue is really the only reason I can't do more than just tinker with moose for fun. This isn't just a moosefs problem, even ceph sends data unencrypted unfortunately.

**Comment by chogata** [comment]
Yes, it's still on the roadmap, but definitely not before some other features.
